ipaserver/install/server/upgrade.py

```diff
@@ -1485,14 +1485,20 @@ def add_default_caacl(ca):
 def setup_pkinit(krb):
     root_logger.info("[Setup PKINIT]")
 
-    if not api.Command.ca_is_enabled()['result']:
-        root_logger.info("CA is not enabled")
-        return
+    pkinit_state = krbinstance.get_local_pkinit_state()
+
+    if pkinit_state not in (krbinstance.PKINIT_FULL,
+                            krbinstance.PKINIT_EXTERNAL):
 
-    if not os.path.exists(paths.KDC_CERT):
-        root_logger.info("Requesting PKINIT certificate")
-        krb.setup_pkinit()
+        pkinit_setup_handler = krb.setup_local_pkinit
 
+        if api.Command.ca_is_enabled()['result']:
+            pkinit_setup_handler = krb.setup_full_pkinit
```

Comment:
If full / external pkinit was not enabled before, we should not enable it during upgrade, otherwise `--no-pkinit` will work only until the first upgrade.